x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/integrations/install_ai_prompts.ts

```diff
@@ -5,19 +5,24 @@
  * 2.0.
  */
 
+import type { Logger } from '@kbn/core/server';
 import { SECURITY_AI_PROMPTS_PACKAGE_NAME } from '../../../../../../common/detection_engine/constants';
 import type { SecuritySolutionApiRequestHandlerContext } from '../../../../../types';
+import { ensureInstalledPackage } from './ensure_installed_package';
 import { findLatestPackageVersion } from './find_latest_package_version';
 
 export async function installSecurityAiPromptsPackage(
-  context: SecuritySolutionApiRequestHandlerContext
+  context: SecuritySolutionApiRequestHandlerContext,
+  logger: Logger
 ) {
   try {
-    const pkgVersion = await findLatestPackageVersion(context, SECURITY_AI_PROMPTS_PACKAGE_NAME);
-    return context.getInternalFleetServices().packages.ensureInstalledPackage({
-      pkgName: SECURITY_AI_PROMPTS_PACKAGE_NAME,
-      pkgVersion,
-    });
+    const pkgVersion = await findLatestPackageVersion(
+      context,
+      SECURITY_AI_PROMPTS_PACKAGE_NAME,
+      logger
+    );
+
+    return ensureInstalledPackage(context, SECURITY_AI_PROMPTS_PACKAGE_NAME, pkgVersion, logger);
```

Comment:
An exception won't be captured by `catch` if you just return the promise without awaiting it.

```suggestion
    return await ensureInstalledPackage(context, SECURITY_AI_PROMPTS_PACKAGE_NAME, pkgVersion, logger);
```